### PR TITLE
[Yaml] chomp newlines only at the end of YAML documents

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Branch? | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18784 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by theofidry** [comment]
It does 👍

File: src/Symfony/Component/Yaml/Parser.php [hunk 1]
```diff
@@ -634,6 +643,11 @@ private function isCurrentLineComment()
         return '' !== $ltrimmedLine && $ltrimmedLine[0] === '#';
     }
 
+    private function isCurrentLineLastLineInDocument()
```
Comment (javiereguiluz) [inline src/Symfony/Component/Yaml/Parser.php]: Let's "nitpick" a bit: would `isLastDocumentLine()` be a better name for `isCurrentLineLastLineInDocument()`?

Comment (xabbuh) [inline src/Symfony/Component/Yaml/Parser.php]: Not sure, we don't pass any argument so it's not really clear to what the method name refers to imo. Note that the `isCurrentLine` prefix is a common pattern in the parser (e.g. `isCurrentLineComment()`, `isCurrentLineBlank()` and so on).

Comment (javiereguiluz) [inline src/Symfony/Component/Yaml/Parser.php]: OK then.

**Comment by javiereguiluz** [comment]
👍 

Status: reviewed

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**